agent/discovery/cluster.go

```diff
@@ -27,26 +27,26 @@ func (c ClusterDiscovery) GetId() string {
 }
 
 // Execute one iteration of a discovery and store the result in the Consul KVStore.
-func (d ClusterDiscovery) Discover() error {
+func (d ClusterDiscovery) Discover() (string, error) {
 	cluster, err := cluster.NewCluster()
 
 	if err != nil {
-		return err
+		return "", err
```

Comment:
I guess we could return here the error converted to a string. Have a look how you can do this as the errors might be wrapped one on top of each other.
And basically expand this to all the error cases

Reply:
Done! Also to the other cases.

Reply:
Sorry, @aleksei-burlakov ,
You were right here. I don't know why I recommended to send something in the 1st variable in a error case...
It doesn't make sense. You can put back an empty string here

Reply:
**Edit**. Ok, now I understand why I requested this...
We needed to return back a correct message because this is the value used to send the message later.
I will run a complete review now, but something smells strange now